src/resources/metadata/trendingDapps.ts

```diff
@@ -0,0 +1,55 @@
+import { metadataClient } from '@/graphql';
+import { DAppRankingPeriod, DAppStatus, TrendingDAppsQuery } from '@/graphql/__generated__/metadata';
+import { createQueryKey, QueryConfigWithSelect } from '@/react-query';
+import { QueryFunction, useQuery } from '@tanstack/react-query';
+
+const trendingDAppsQueryKey = ({ period }: { period: DAppRankingPeriod }) =>
+  createQueryKey('trendingDApps', { period }, { persisterVersion: 1 });
+
+type TrendingDAppsQueryKey = ReturnType<typeof trendingDAppsQueryKey>;
+
+const fetchTrendingDApps: QueryFunction<TrendingDAppsQuery, TrendingDAppsQueryKey> = async ({ queryKey }) => {
+  const [{ period }] = queryKey;
+
+  return metadataClient.trendingDApps({
+    period,
+  });
+};
+
+export function selectValidDApps(data: TrendingDAppsQuery): TrendingDAppsQuery {
+  if (!data.dApps?.length) return { dApps: [] };
+  return {
+    ...data,
+    dApps: data.dApps.filter(Boolean),
+  };
+}
+
+export function selectNonScamDApps(data: TrendingDAppsQuery): TrendingDAppsQuery {
+  if (!data.dApps?.length) return { dApps: [] };
+  return {
+    ...data,
+    dApps: data.dApps.filter(d => d?.status !== DAppStatus.Scam),
+  };
+}
```

Comment:
Any reason not to combine these two? I think we'll always want to filter out scams.

Reply:
Referring to these:

```typescript
export function selectValidDApps(data: TrendingDAppsQuery): TrendingDAppsQuery {
  if (!data.dApps?.length) return { dApps: [] };
  return {
    ...data,
    dApps: data.dApps.filter(Boolean),
  };
}

export function selectNonScamDApps(data: TrendingDAppsQuery): TrendingDAppsQuery {
  if (!data.dApps?.length) return { dApps: [] };
  return {
    ...data,
    dApps: data.dApps.filter(d => d?.status !== DAppStatus.Scam),
  };
}
```

Reply:
We can combine then sure. I was trying to keep selectors as narrow as possible.